src/main/java/com/minecolonies/entity/ai/citizen/farmer/FieldView.java

```diff
@@ -0,0 +1,129 @@
+package com.minecolonies.entity.ai.citizen.farmer;
+
+import com.minecolonies.util.BlockPosUtil;
+import com.minecolonies.util.BlockUtils;
+import io.netty.buffer.ByteBuf;
+import net.minecraft.item.Item;
+import net.minecraft.util.BlockPos;
+import net.minecraftforge.fml.common.network.ByteBufUtils;
+
+/**
+ * Client side representation of a field.
+ */
+public class FieldView
+{
+    /**
+     * Unique id of the field.
+     */
+    private BlockPos id;
+    private boolean  taken;
+    private String   owner;
+    private Item     item;
+
+    /**
+     * Constructor to instantiate field to fill it from a byteBuffer.
+     */
+    public FieldView()
+    {
+        /**
+         * Intentionally left empty.
+         */
+    }
+
+    /**
+     * Create the fieldView.
+     *
+     * @param field the field.
+     */
+    public FieldView(Field field)
+    {
+        this.id = field.getID();
+        this.taken = field.isTaken();
+        this.owner = field.getOwner();
+        this.item = field.getSeed();
+    }
+
+    /**
+     * Getter of the field id.
+     * @return blockPos of the field.
+     */
+    public BlockPos getId()
+    {
+        return id;
+    }
+
+    /**
+     * Checks if field has been claimed yet.
+     * @return true if so.
+     */
+    public boolean isTaken()
+    {
+        return taken;
+    }
+
+    /**
+     * Checks for the name of the owner.
+     * @return String of the name.
+     */
+    public String getOwner()
+    {
+        return owner;
+    }
+
+    /**
+     * Claims or frees the field.
+     * @param taken true if should be claimed.
+     */
+    public void setTaken(final boolean taken)
+    {
+        this.taken = taken;
+    }
+
+    /**
+     * Sets the owner of the field.
+     * @param owner name of the owner.
+     */
+    public void setOwner(final String owner)
+    {
+        this.owner = owner;
+    }
+
+    /**
+     * Writes the fields data to a byte buf for transition.
+     *
+     * @param buf Buffer to write to.
+     */
+    public void serializeViewNetworkData(ByteBuf buf)
+    {
+        BlockPosUtil.writeToByteBuf(buf, id);
+        buf.writeBoolean(taken);
+        ByteBufUtils.writeUTF8String(buf, owner);
+        int itemId = item == null? 0 : Item.getIdFromItem(item);
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Local variable 'itemId' could be declared final [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://home.kk-sc.de:9000/coding_rules#rule_key=pmd%3ALocalVariableCouldBeFinal)